OpenRA.Game/Scripting/ScriptTypes.cs

```diff
@@ -149,23 +148,25 @@ public static bool TryGetClrValue(this LuaValue value, Type t, out object clrObj
 
 		public static LuaValue ToLuaValue(this object obj, ScriptContext context)
 		{
-			if (obj is LuaValue)
-				return (LuaValue)obj;
+			{
```

Comment:
I inserted a new block scope intentionally. All the new variable names polluted the later scope and it seemed cleanist to just contain the scope since we don't care about the variables if the match fails. Specifically without a new scope or a name change the `obj is int i` aliases the loop variable `var i = 1;` later in the method.